src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/WorkspaceProject.cs

```diff
@@ -170,8 +170,7 @@ public async Task SetBuildSystemPropertiesAsync(IReadOnlyDictionary<string, stri
             }
         }
 
-        // Workaround for https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1830960
-        _project.CompilationOutputAssemblyFilePath = _project.OutputFilePath;
+        _project.CompilationOutputAssemblyFilePath = _project.CompilationOutputAssemblyFilePath;
```

Comment:
Bug has been resolved so can use correct information from devkit

Reply:
This line doesn't do anything.

Also this method seems to be missing the case where it actually sets the `CompilationOutputAssemblyFilePath` info it gets from the underlying project system: https://github.com/dotnet/roslyn/pull/70913/files#diff-af31a9956c1c1a4a99e36eb99ec37faa9b547e01645b4b9a36c4abc24e0f1476R149

Reply:
Also missing this bug fix: https://github.com/dotnet/roslyn/pull/70913/files#r1400060227

Reply:
this is reassigning to itself, do you need to get the value from somewhere else (is it supposed to come from the properties parameter)?

Reply:
Oh woops, yup. Need to have `case "IntermediateAssembly": _project.CompilationOutputAssemblyFilePath = GetFullyQualifiedPath(valueOrNull); break;`